### ISSUE TITLE
[Feat]: a11y enhancement visual list for sp-field-group

### ISSUE THREAD

**Issue by ccheung0926** [adobe/spectrum-web-components#5697]
### Code of conduct

- [x] I agree to follow this project's code of conduct.

### Impacted component(s)

sp-field-group

### Description of the requested feature

According to [Adobe storybook](https://opensource.adobe.com/spectrum-web-components/storybook/index.html?path=/docs/field-group--docs), sp-field-group should be using `ul li` list rather than `role="group" for screen reader user.

you can go to this link to check the checkbox items: https://new.express.adobe.com/explore/templates 

<img width="238" height="292" alt="Image" src="https://github.com/user-attachments/assets/e838b543-c07a-493a-bcae-fed1c2fd639d" />

### Would you like to track this issue in Jira?

- [ ] Yes, please tell me the ticket number!
